Src/DependencyCollector/Shared/Implementation/HttpCoreDiagnosticSourceListener.cs

```diff
@@ -300,6 +300,10 @@ private void InjectRequestHeaders(HttpRequestMessage request, string instrumenta
                             {
                                 HttpHeadersUtilities.SetRequestContextKeyValue(requestHeaders, RequestResponseHeaders.RequestContextCorrelationSourceKey, sourceApplicationId);
                             }
+                            else
```

Comment:
There we create a new implication that when TryGetXComponentCorrelationId returns false, it means it is empty appId, right? I am a bit worried about this assumption because there could be other cases. For example, when an invalid configuration leads to request failure. Can we check the header again to making sure it is the case of empty appId being returned?